onnx/helper.py

```diff
@@ -854,34 +857,57 @@ def make_attribute(  # pylint: disable=too-many-statements
     # Iterable cases
     elif isinstance(value, collections.abc.Iterable):
         value = list(value)
-        types = {type(v) for v in value}
-        if all(issubclass(t, numbers.Integral) for t in types):
+        if len(value) == 0 and attr_type is None:
+            raise ValueError("Could not infer attribute type from empty iterator")
+        if attr_type is None:
+            types = {type(v) for v in value}
```

Comment:
After this line, you can check the length of types and raise an exception if it is >= 2 instead of check all types in this list are the same.